pkg/cloudprovider/types.go

```diff
@@ -31,23 +31,20 @@ type CloudProvider interface {
 	// requests. The callback must be called with a theoretical node object that
 	// is fulfilled by the cloud providers capacity creation request. This API
 	// is called in parallel and then waits for all channels to return nil or error.
-	Create(context.Context, *v1alpha3.Provisioner, *v1alpha3.Constraints, []InstanceType, func(*v1.Node) error) chan error
+	Create(context.Context, *v1alpha4.Constraints, []InstanceType, func(*v1.Node) error) chan error
+	// Delete node in cloudprovider
+	Delete(context.Context, *v1.Node) error
```

Comment:
Minor comment for discussion - wonder if this interface should be split into multiple? (I realize there are pros and cons). It just seems like Create/Delete are very node-specific and action-oriented, then GetInstanceTypes/GetZones seem more like utility functions, more oriented around returning metadata (about constraints) to some higher-level code?

Reply:
I'm interested. I'd like to see how much impact it has to code bloat. It's fairly standard to have a set of methods like this on a cloudprovider (from what I've seen in other projects). I agree with you conceptually, though.